// inference.proto
syntax = "proto3";
import "common.proto";

package trueno;

service Inference {
  rpc CreateAndLoadModel (LoadRequest) returns (ResultReply) {}
  rpc ReloadModel (LoadRequest) returns (ResultReply) {}
  rpc InferenceLocal (InferRequest) returns (ResultReply) {}
  rpc InferenceRemote (InferRequest) returns (ResultReply) {}
}

message LoadRequest {
  string bid = 1;
  string btype = 2;
  string model = 3;
  string version = 4;
  string mode = 5;
  string extra = 6;
}

message InferRequest {
  string bid = 1;
  string uuid = 2;
  string path = 3;
  string type = 4;
  string base64 = 5;
}

